src/components/utils/sample.ts

```diff
@@ -0,0 +1,128 @@
+import yaml from 'js-yaml';
+
+function sampleCode(code: string, type: string, token1: string, token2: string): string {
+    if (type == "protobuf") {
+        return sampleProto(code, token1, token2);
+    } else if (type == "yml") {
+        return sampleYaml(code, token1);
+    }
+    throw new Error(`Type not found: ${type}`);
+}
+
+function sampleYaml(code: string, path: string): string {
+    const yaml_code = yaml.load(code);
+
+    // Convert dot notation path into nested object
+    let segments = path.split('.');
+    let result = {};
+    let current = result;
+
+    let traversal = yaml_code;
+
+    for (let i = 0; i < segments.length - 1; i++) {
+        if (segments[i] != "$") {
+            current[segments[i]] = {};
+            current = current[segments[i]];
+
+            if(!(segments[i] in traversal)) {
+                throw new Error(`Invalid JSON Path: ${path}`);
+            }
+            traversal = traversal[segments[i]];
+        }
+    }
+    current[segments[segments.length - 1]] = traversal;
+
+    return yaml.dump(result);
+}
+
+function sampleProto(code: string, token1: string, token2: string) {
```

Comment:
proto code looks pretty hairy.... I wonder if there is some sort of protobuf parser we can rely on at some point?

the syntax is much, much more complex than OpenAPI though.

Reply:
We aren't actually doing any protobuf parsing. The syntax we've chosen revolves around looking at indentation levels. 

I really dislike this, so I'll take a look at it.